### PR TITLE
RadiusGraph used to compute neighbors

### PR DESCRIPTION
**Pull request by allaffa**

### CODE REVIEW

**Comment by allaffa** [comment]
@streeve @pzhanggit 

Please, review this at your earliest convenience. I need to replace hard-coded transformations in PR# as well for the edge length, and I want to merge this PR first so that I can rebase the other one. Since I am touching the same portion of the code, I want to make my life as easy as possible and avoid conflicts.

File: requirements-torchdep.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 torch-geometric>=1.7.2 #required for GATv2
 torch-scatter
 torch-sparse
+torch-cluster
```
Comment (streeve) [inline requirements-torchdep.txt:4]: Do transforms in general require torch_cluster?

File: hydragnn/preprocess/serialized_dataset_loader.py [hunk 2]
```diff
@@ -66,15 +67,14 @@ def load_serialized_data(self, dataset_path: str, config):
             _ = pickle.load(f)
             dataset = pickle.load(f)
 
-        edge_index, edge_distances = self.__compute_edges(
-            data=dataset[0],
-            radius=config["Architecture"]["radius"],
-            max_num_node_neighbours=config["Architecture"]["max_neighbours"],
+        compute_edges = RadiusGraph(
```
Comment (streeve) [inline hydragnn/preprocess/serialized_dataset_loader.py:70]: This is a very nice change. I'll have to think through how to add periodic edges though

File: requirements-torchdep.txt (hunk 1 above)
Comment (allaffa) [reply requirements-torchdep.txt:4]: RadiusGraph requires it, and Radius Graph is needed to build the edges.

File: hydragnn/preprocess/serialized_dataset_loader.py (hunk 2 above)
Comment (allaffa) [reply hydragnn/preprocess/serialized_dataset_loader.py:70]: I am going to think about it too.

File: hydragnn/preprocess/serialized_dataset_loader.py [hunk 3]
```diff
@@ -127,71 +127,6 @@ def __update_predicted_values(self, type: list, index: list, data: Data):
...
-            self.verbosity, "Computing edge distances and adding candidate neighbours."
-        )
-        for i in iterate_tqdm(range(num_of_atoms), self.verbosity):
-            for j in range(num_of_atoms):
-                distance = distance_3D(data.pos[i], data.pos[j])
-                distance_matrix[i, j] = distance
-                if distance_matrix[i, j] <= radius and i != j:
-                    candidate_neighbours[i].append(j)
-
-        candidate_neighbours = order_candidates(
```
Comment (streeve) [inline hydragnn/preprocess/serialized_dataset_loader.py:165]: Should we remove the unused util functions? Or do you think we may have other uses for them?

Comment (allaffa) [reply hydragnn/preprocess/serialized_dataset_loader.py:165]: Nice catch!
Those functions are not used now.
I would rather remove3 them for the time being.

File: hydragnn/preprocess/helper_functions.py [hunk 4]
```diff
@@ -14,101 +14,6 @@
...
-    if len(already_neighbours) == 0:
-        return neighbours[:max_num_node_neighbours]
-
-    number_of_allowed = max_num_node_neighbours - len(already_neighbours)
-    neighbours = neighbours[:number_of_allowed] + already_neighbours
-
-    return neighbours
-
 
 def tensor_divide(x1, x2):
```
Comment (streeve) [inline hydragnn/preprocess/helper_functions.py:18]: Let's just put this in `raw_dataset_loader` until we need it elsewhere

**Review by streeve — APPROVED** [review]

**Pull request merged**